<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="TransProPy is a Python package that is part of the TransPro series under the SuperOmics framework.">

<title>TransProPyBook - 7&nbsp; AutoGluonSelectML.py</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./AutoFeatureSelection.html" rel="next">
<link href="./AutogluonTimeLimit.html" rel="prev">
<link href="./TransProPy_logo.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="custom.css">
<meta property="og:title" content="TransProPyBook - 7&nbsp; AutoGluonSelectML.py">
<meta property="og:description" content="Python package TransProPy.">
<meta property="og:image" content="https://github.com/SSSYDYSSS/TransProPyBook/blob/main/TransProPy_logo.png">
<meta property="og:site-name" content="TransProPyBook">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./AutoGluonSelectML.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">AutoGluonSelectML.py</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./TransProPy_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">TransProPyBook</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/SSSYDYSSS/TransProPyBook" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <div class="dropdown">
      <a href="" title="Share" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Share"><i class="bi bi-share"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
              <i class="bi bi-bi-twitter pe-1"></i>
            Twitter
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=|url|">
              <i class="bi bi-bi-facebook pe-1"></i>
            Facebook
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://www.linkedin.com/sharing/share-offsite/?url=|url|">
              <i class="bi bi-bi-linkedin pe-1"></i>
            LinkedIn
            </a>
          </li>
      </ul>
    </div>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./UtilsFunction1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">UtilsFunction1</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./UtilsFunction2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">UtilsFunction2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./UtilsFunction3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">UtilsFunction3</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./MACFCmain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">MACFCmain.py</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./NewMACFCmain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">NewMACFCmain.py</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./AutogluonTimeLimit.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">AutogluonTimeLimit.py</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./AutoGluonSelectML.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">AutoGluonSelectML.py</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./AutoFeatureSelection.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">AutoFeatureSelection.py</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./funkyheatmap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Comparison of TransPropy with Other Tool Packages Using Funkyheatmap</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Percentage Stacked Line Bar Chart.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Comparison of TransPropy with Other Tool Packages Using Percentage Stacked Line Bar Chart</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Ridge Plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Comparison of TransPropy with Other Tool Packages Using Ridge Plot</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CFD FullGSEA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Comparison of TransPropy with Other Tool Packages Using GSEA (Gene: CFD)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ANKRD35 FullGSEA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Comparison of TransPropy with Other Tool Packages Using GSEA (Gene: ANKRD35)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ALOXE3 FullGSEA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Comparison of TransPropy with Other Tool Packages Using GSEA (Gene: ALOXE3)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./GeneCount and GseaNES.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Comparison of TransPropy with Other Tool Packages Using GeneCount and GseaNES Data (Gene: CFD/ANKRD35/ALOXE3)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ALL GENE LINE BAR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">All gene line bar</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Seprate All GSEA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Comparison of TransPropy with Other Tool Packages Using Seprate All GSEA (Gene: CFD/ANKRD35/ALOXE3)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./VennNetwork core enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Comparison of TransPropy with Other Tool Packages Using VennNetwork with core enrichment (Gene: CFD/ANKRD35/ALOXE3)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CircosBar core enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Comparison of TransPropy with Other Tool Packages Using CircosBar</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Contact.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Contact</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">References</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#parameters-d" id="toc-parameters-d" class="nav-link active" data-scroll-target="#parameters-d"><span class="header-section-number">7.1</span> Parameters</a></li>
  <li><a href="#returns-d" id="toc-returns-d" class="nav-link" data-scroll-target="#returns-d"><span class="header-section-number">7.2</span> Return</a></li>
  <li><a href="#usage-of-autogluon_selectml" id="toc-usage-of-autogluon_selectml" class="nav-link" data-scroll-target="#usage-of-autogluon_selectml"><span class="header-section-number">7.3</span> Usage of Autogluon_SelectML</a>
  <ul class="collapse">
  <li><a href="#objectives" id="toc-objectives" class="nav-link" data-scroll-target="#objectives"><span class="header-section-number">7.3.1</span> Objectives</a></li>
  <li><a href="#note" id="toc-note" class="nav-link" data-scroll-target="#note"><span class="header-section-number">7.3.2</span> Note</a></li>
  </ul></li>
  <li><a href="#insignificant-correlation-d" id="toc-insignificant-correlation-d" class="nav-link" data-scroll-target="#insignificant-correlation-d"><span class="header-section-number">7.4</span> Insignificant Correlation</a>
  <ul class="collapse">
  <li><a href="#import-the-corresponding-module" id="toc-import-the-corresponding-module" class="nav-link" data-scroll-target="#import-the-corresponding-module"><span class="header-section-number">7.4.1</span> Import the corresponding module</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">7.4.2</span> Data</a></li>
  <li><a href="#autogluon_selectml" id="toc-autogluon_selectml" class="nav-link" data-scroll-target="#autogluon_selectml"><span class="header-section-number">7.4.3</span> Autogluon_SelectML</a></li>
  <li><a href="#return" id="toc-return" class="nav-link" data-scroll-target="#return"><span class="header-section-number">7.4.4</span> Return</a></li>
  <li><a href="#save-data" id="toc-save-data" class="nav-link" data-scroll-target="#save-data"><span class="header-section-number">7.4.5</span> Save Data</a></li>
  </ul></li>
  <li><a href="#significant-correlation-d" id="toc-significant-correlation-d" class="nav-link" data-scroll-target="#significant-correlation-d"><span class="header-section-number">7.5</span> Significant Correlation</a>
  <ul class="collapse">
  <li><a href="#import-the-corresponding-module-1" id="toc-import-the-corresponding-module-1" class="nav-link" data-scroll-target="#import-the-corresponding-module-1"><span class="header-section-number">7.5.1</span> Import the corresponding module</a></li>
  <li><a href="#data-1" id="toc-data-1" class="nav-link" data-scroll-target="#data-1"><span class="header-section-number">7.5.2</span> Data</a></li>
  <li><a href="#autogluon_selectml-1" id="toc-autogluon_selectml-1" class="nav-link" data-scroll-target="#autogluon_selectml-1"><span class="header-section-number">7.5.3</span> Autogluon_SelectML</a></li>
  <li><a href="#return-1" id="toc-return-1" class="nav-link" data-scroll-target="#return-1"><span class="header-section-number">7.5.4</span> Return</a></li>
  <li><a href="#save-data-1" id="toc-save-data-1" class="nav-link" data-scroll-target="#save-data-1"><span class="header-section-number">7.5.5</span> Save Data</a></li>
  </ul></li>
  <li><a href="#references-d" id="toc-references-d" class="nav-link" data-scroll-target="#references-d"><span class="header-section-number">7.6</span> References</a>
  <ul class="collapse">
  <li><a href="#scientific-publications" id="toc-scientific-publications" class="nav-link" data-scroll-target="#scientific-publications"><span class="header-section-number">7.6.1</span> Scientific Publications</a></li>
  <li><a href="#articles" id="toc-articles" class="nav-link" data-scroll-target="#articles"><span class="header-section-number">7.6.2</span> Articles</a></li>
  <li><a href="#documentation" id="toc-documentation" class="nav-link" data-scroll-target="#documentation"><span class="header-section-number">7.6.3</span> Documentation</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/SSSYDYSSS/TransProPyBook/edit/main/AutoGluonSelectML.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/SSSYDYSSS/TransProPyBook/issues/new" class="toc-action">Report an issue</a></p><p><a href="https://github.com/SSSYDYSSS/TransProPyBook/blob/main/AutoGluonSelectML.qmd" class="toc-action">View source</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">AutoGluonSelectML.py</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Trains a model using AutoGluon on provided data path and returns feature importance and model leaderboard.</p>
<section id="parameters-d" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="parameters-d"><span class="header-section-number">7.1</span> Parameters</h2>
<blockquote class="blockquote">
<ul>
<li><b style="color: #43799a;">gene_data_path</b> (str):
<ul>
<li>Path to the gene expression data CSV file.</li>
<li><b style="color: #54467d;">For example</b>: ‘../data/gene_tpm.csv’</li>
</ul></li>
<li><b style="color: #43799a;">class_data_path</b> (str):
<ul>
<li>Path to the class data CSV file.</li>
<li><b style="color: #54467d;">For example</b>: ‘../data/tumor_class.csv’</li>
</ul></li>
<li><b style="color: #43799a;">label_column</b> (str):
<ul>
<li>Name of the column in the dataset that is the target label for prediction.</li>
</ul></li>
<li><b style="color: #43799a;">test_size</b> (float):
<ul>
<li>Proportion of the data to be used as the test set.</li>
</ul></li>
<li><b style="color: #43799a;">threshold</b> (float):
<ul>
<li>The threshold used to filter out rows based on the proportion of non-zero values.</li>
</ul></li>
<li><b style="color: #43799a;">hyperparameters</b> (dict, <b style="color: #406e1a;">optional</b>):
<ul>
<li>Dictionary of hyperparameters for the models.</li>
<li><b style="color: #54467d;">For example</b>: {‘GBM’: {}, ‘RF’: {}}</li>
</ul></li>
<li><b style="color: #43799a;">random_feature</b> (int, <b style="color: #406e1a;">optional</b>):
<ul>
<li>The number of random feature to select. If None, no random feature selection is performed.</li>
<li><b style="color: #406e1a;">Default is None</b>.</li>
</ul></li>
<li><b style="color: #43799a;">num_bag_folds</b> (int, <b style="color: #406e1a;">optional</b>)
<ul>
<li><b style="color: #54467d;">Please note</b>: This parameter annotation source can be referred to the documentation link in References.</li>
<li>Number of folds used for bagging of models. When <code>num_bag_folds = k</code>, training time is roughly increased by a factor of <code>k</code> (set = 0 to disable bagging). Disabled by default (0), but we recommend values between 5-10 to maximize predictive performance. Increasing <code>num_bag_folds</code> will result in models with lower bias but that are more prone to overfitting. <code>num_bag_folds = 1</code> is an invalid value, and will raise a ValueError. Values &gt; 10 may produce diminishing returns, and can even harm overall results due to overfitting. To further improve predictions, avoid increasing <code>num_bag_folds</code> much beyond 10 and instead increase <code>num_bag_sets</code>.</li>
<li><b style="color: #406e1a;">default = None</b></li>
</ul></li>
<li><b style="color: #43799a;">num_stack_levels</b> (int, <b style="color: #406e1a;">optional</b>)
<ul>
<li><b style="color: #54467d;">Please note</b>: This parameter annotation source can be referred to the documentation link in References.</li>
<li>Number of stacking levels to use in stack ensemble. Roughly increases model training time by factor of <code>num_stack_levels+1</code> (set = 0 to disable stack ensembling). Disabled by default (0), but we recommend values between 1-3 to maximize predictive performance. To prevent overfitting, <code>num_bag_folds &gt;= 2</code> must also be set or else a ValueError will be raised.</li>
<li><b style="color: #406e1a;">default = None</b></li>
</ul></li>
<li><b style="color: #43799a;">time_limit</b> (int, <b style="color: #406e1a;">optional</b>):
<ul>
<li>Time limit for training in seconds.</li>
<li><b style="color: #406e1a;">default is 120</b>.</li>
</ul></li>
<li><b style="color: #43799a;">random_state</b> (int, <b style="color: #406e1a;">optional</b>):
<ul>
<li>The seed used by the random number generator.</li>
<li><b style="color: #406e1a;">default is 42</b>.</li>
</ul></li>
</ul>
</blockquote>
</section>
<section id="returns-d" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="returns-d"><span class="header-section-number">7.2</span> Return</h2>
<blockquote class="blockquote">
<ul>
<li><b style="color: #43799a;">importance</b> (DataFrame):
<ul>
<li>DataFrame containing feature importance.</li>
</ul></li>
<li><b style="color: #43799a;">leaderboard</b> (DataFrame):
<ul>
<li>DataFrame containing model performance on the test data.</li>
</ul></li>
</ul>
</blockquote>
</section>
<section id="usage-of-autogluon_selectml" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="usage-of-autogluon_selectml"><span class="header-section-number">7.3</span> Usage of Autogluon_SelectML</h2>
<p>Performing training and prediction tasks on tabular data using Autogluon.</p>
<section id="objectives" class="level3" data-number="7.3.1">
<h3 data-number="7.3.1" class="anchored" data-anchor-id="objectives"><span class="header-section-number">7.3.1</span> Objectives</h3>
<section id="model-training-and-selection" class="level4" data-number="7.3.1.1">
<h4 data-number="7.3.1.1" class="anchored" data-anchor-id="model-training-and-selection"><span class="header-section-number">7.3.1.1</span> Model Training and Selection</h4>
<blockquote class="blockquote">
<p>Autogluon will attempt various models and hyperparameter combinations within a given time limit to find the best-performing model on the test data. During training, Autogluon may output training logs displaying performance metrics and progress information for different models. The goal is to select the best-performing model for use in subsequent prediction tasks.</p>
</blockquote>
</section>
<section id="leaderboard" class="level4" data-number="7.3.1.2">
<h4 data-number="7.3.1.2" class="anchored" data-anchor-id="leaderboard"><span class="header-section-number">7.3.1.2</span> Leaderboard</h4>
<blockquote class="blockquote">
<p>The leaderboard displays performance scores of different models on the test data, typically including metrics like accuracy, precision, recall, and more. The purpose is to assist users in understanding the performance of different models to choose the most suitable model for predictions.</p>
</blockquote>
</section>
<section id="importance" class="level4" data-number="7.3.1.3">
<h4 data-number="7.3.1.3" class="anchored" data-anchor-id="importance"><span class="header-section-number">7.3.1.3</span> Importance</h4>
<blockquote class="blockquote">
<p>Feature importance indicates which features are most critical for the model’s prediction performance. The purpose is to help users understand the importance of specific features in the data, which can be used for feature selection or further data analysis.</p>
</blockquote>
</section>
</section>
<section id="note" class="level3" data-number="7.3.2">
<h3 data-number="7.3.2" class="anchored" data-anchor-id="note"><span class="header-section-number">7.3.2</span> Note</h3>
<blockquote class="blockquote">
<p>Please note that Autogluon’s output results may vary depending on your data and task. You can review the generated model leaderboard and feature importance to understand model performance and the significance of specific features in the data. These results can aid you in making better predictions and decisions.</p>
</blockquote>
</section>
</section>
<section id="insignificant-correlation-d" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="insignificant-correlation-d"><span class="header-section-number">7.4</span> Insignificant Correlation</h2>
<blockquote class="blockquote">
<ul>
<li><b style="color: #54467d;">Please note</b>:Data characteristics: Features have weak correlation with the classification.</li>
<li>Randomly shuffling the class labels to a certain extent simulates reducing the correlation.</li>
</ul>
</blockquote>
<section id="import-the-corresponding-module" class="level3" data-number="7.4.1">
<h3 data-number="7.4.1" class="anchored" data-anchor-id="import-the-corresponding-module"><span class="header-section-number">7.4.1</span> Import the corresponding module</h3>
<div class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> TransProPy.AutogluonSelectML <span class="im">import</span> AutoGluon_SelectML</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
</section>
<section id="data" class="level3" data-number="7.4.2">
<h3 data-number="7.4.2" class="anchored" data-anchor-id="data"><span class="header-section-number">7.4.2</span> Data</h3>
<div class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>data_path <span class="op">=</span> <span class="st">'../test_TransProPy/data/four_methods_degs_intersection.csv'</span>  </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> pd.read_csv(data_path)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(data.iloc[:<span class="dv">10</span>, :<span class="dv">10</span>]) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Unnamed: 0  TCGA-D9-A4Z2-01A  TCGA-ER-A2NH-06A  TCGA-BF-A5EO-01A  \
0        A2M         16.808499         16.506184         17.143433   
1      A2ML1          1.584963          9.517669          7.434628   
2      AADAC          4.000000          2.584963          1.584963   
3    AADACL2          1.000000          1.000000          0.000000   
4     ABCA12          4.523562          4.321928          3.906891   
5    ABCA17P          4.584963          5.169925          3.807355   
6      ABCA9          9.753217          6.906891          3.459432   
7      ABCB4          9.177420          6.700440          5.000000   
8      ABCB5         10.134426          4.169925          9.167418   
9     ABCC11         10.092757          6.491853          5.459432   

   TCGA-D9-A6EA-06A  TCGA-D9-A4Z3-01A  TCGA-GN-A26A-06A  TCGA-D3-A3BZ-06A  \
0         17.760739         14.766839         16.263691         16.035207   
1          2.584963          1.584963          2.584963          5.285402   
2          0.000000          0.000000          0.000000          3.321928   
3          0.000000          1.000000          0.000000          0.000000   
4          3.459432          1.584963          3.000000          4.321928   
5          8.366322          7.228819          7.076816          4.584963   
6          2.584963          6.357552          6.475733          7.330917   
7          9.342075         10.392317          7.383704         11.032735   
8          4.906891         11.340963          3.169925         11.161762   
9          6.807355          4.247928          5.459432          5.977280   

   TCGA-D3-A51G-06A  TCGA-EE-A29R-06A  
0         18.355114         16.959379  
1          2.584963          3.584963  
2          1.000000          4.584963  
3          0.000000          1.000000  
4          4.807355          3.700440  
5          6.409391          7.139551  
6          7.954196          9.177420  
7         10.082149         10.088788  
8          4.643856         12.393927  
9          5.614710          8.233620  </code></pre>
</div>
</div>
<hr>
<div class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>data_path <span class="op">=</span> <span class="st">'../test_TransProPy/data/random_classification_class.csv'</span>  </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> pd.read_csv(data_path)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(data.iloc[:<span class="dv">10</span>, :<span class="dv">10</span>]) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         Unnamed: 0  class
0  TCGA-D9-A4Z2-01A      2
1  TCGA-ER-A2NH-06A      2
2  TCGA-BF-A5EO-01A      2
3  TCGA-D9-A6EA-06A      2
4  TCGA-D9-A4Z3-01A      1
5  TCGA-GN-A26A-06A      1
6  TCGA-D3-A3BZ-06A      1
7  TCGA-D3-A51G-06A      1
8  TCGA-EE-A29R-06A      1
9  TCGA-D3-A2JE-06A      1</code></pre>
</div>
</div>
<hr>
</section>
<section id="autogluon_selectml" class="level3" data-number="7.4.3">
<h3 data-number="7.4.3" class="anchored" data-anchor-id="autogluon_selectml"><span class="header-section-number">7.4.3</span> Autogluon_SelectML</h3>
<blockquote class="blockquote">
<ul>
<li>The core purpose of choosing Autogluon_SelectML — to select a larger feature set in AutoGluon that includes both important and secondary features — is reflected in the following custom hyperparameters configuration. This setup is designed to utilize multiple model types so that the models can consider a broader range of features.</li>
<li>This configuration encompasses neural networks (using PyTorch and FastAI), gradient boosting machines (LightGBM, XGBoost, and CatBoost), random forests (RF), extremely randomized trees (XT), K-nearest neighbors (KNN), and linear regression (LR).</li>
</ul>
</blockquote>
<div class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>importance, leaderboard <span class="op">=</span> AutoGluon_SelectML(</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>    gene_data_path<span class="op">=</span><span class="st">'../test_TransProPy/data/four_methods_degs_intersection.csv'</span>, </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>    class_data_path<span class="op">=</span><span class="st">'../test_TransProPy/data/random_classification_class.csv'</span>, </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>    label_column<span class="op">=</span><span class="st">'class'</span>, </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    test_size<span class="op">=</span><span class="fl">0.3</span>, </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>    threshold<span class="op">=</span><span class="fl">0.9</span>, </span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>    hyperparameters<span class="op">=</span>{</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">'GBM'</span>: {}, </span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">'RF'</span>: {},</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">'CAT'</span>: {}, </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">'XGB'</span> : {},</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 'NN_TORCH': {}, </span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 'FASTAI': {},</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">'XT'</span>: {}, </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">'KNN'</span>: {},</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">'LR'</span>: {}</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>        },</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>    random_feature<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>    num_bag_folds<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>    num_stack_levels<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>    time_limit<span class="op">=</span><span class="dv">1000</span>, </span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>    random_state<span class="op">=</span><span class="dv">42</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>No path specified. Models will be saved in: "AutogluonModels\ag-20240804_045523\"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Beginning AutoGluon training ... Time limit = 1000s</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>AutoGluon will save models to "AutogluonModels\ag-20240804_045523\"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>AutoGluon Version:  0.8.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Python Version:     3.10.11</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Operating System:   Windows</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Platform Machine:   AMD64</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Platform Version:   10.0.19044</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Disk Space Avail:   208.49 GB / 925.93 GB (22.5%)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Train Data Rows:    896</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Train Data Columns: 1605</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Label Column: class</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Preprocessing data ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>AutoGluon infers your prediction problem is: 'binary' (because only two unique label-values observed).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    2 unique label values:  [1, 2]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    If 'binary' is not the correct problem_type, please manually specify the problem_type parameter during predictor init (You may specify problem_type as one of: ['binary', 'multiclass', 'regression'])</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Selected class &lt;--&gt; label mapping:  class 1 = 2, class 0 = 1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Note: For your binary classification, AutoGluon arbitrarily selected which label-value represents positive (2) vs negative (1) class.
    To explicitly set the positive_class, either rename classes to 1 and 0, or specify positive_class in Predictor init.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using Feature Generators to preprocess the data ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting AutoMLPipelineFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Available Memory:                    13766.09 MB</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Train Data (Original)  Memory Usage: 11.5 MB (0.1% of available memory)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Inferring data type of each feature based on column values. Set feature_metadata_in to manually specify special dtypes of the features.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Stage 1 Generators:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        Fitting AsTypeFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Stage 2 Generators:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        Fitting FillNaFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Stage 3 Generators:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        Fitting IdentityFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Stage 4 Generators:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        Fitting DropUniqueFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Stage 5 Generators:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        Fitting DropDuplicatesFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Types of features in original data (raw dtype, special dtypes):</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        ('float', []) : 1605 | ['A2M', 'A2ML1', 'ABCA12', 'ABCA17P', 'ABCA9', ...]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Types of features in processed data (raw dtype, special dtypes):</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        ('float', []) : 1605 | ['A2M', 'A2ML1', 'ABCA12', 'ABCA17P', 'ABCA9', ...]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.3s = Fit runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1605 features in original data used to generate 1605 features in processed data.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Train Data (Processed) Memory Usage: 11.5 MB (0.1% of available memory)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Data preprocessing and feature engineering runtime = 1.37s ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>AutoGluon will gauge predictive performance using evaluation metric: 'accuracy'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    To change this, specify the eval_metric parameter of Predictor()</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Automatically generating train/validation split with holdout_frac=0.2, Train Rows: 716, Val Rows: 180</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>User-specified model hyperparameters to be fit:
{
    'GBM': {},
    'RF': {},
    'CAT': {},
    'XGB': {},
    'XT': {},
    'KNN': {},
    'LR': {},
}</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting 7 L1 models ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: KNeighbors ... Training model for up to 998.63s of the 998.62s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.5722   = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.09s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.14s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: LightGBM ... Training model for up to 997.39s of the 997.37s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.6667   = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    9.65s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.01s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: RandomForest ... Training model for up to 987.68s of the 987.66s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.6278   = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    3.84s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.06s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: CatBoost ... Training model for up to 983.73s of the 983.71s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.6667   = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    67.73s   = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.04s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: ExtraTrees ... Training model for up to 915.93s of the 915.91s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.6167   = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.25s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.06s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: XGBoost ... Training model for up to 914.57s of the 914.55s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.6778   = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    26.02s   = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.02s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: LinearModel ... Training model for up to 888.5s of the 888.48s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>E:\Anaconda\Anaconda\envs\TransPro\lib\site-packages\sklearn\preprocessing\_data.py:2663: UserWarning:

n_quantiles (1000) is greater than the total number of samples (716). n_quantiles is set to n_samples.
</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.6  = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    3.03s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.07s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: WeightedEnsemble_L2 ... Training model for up to 360.0s of the 885.33s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.6833   = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.41s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.0s     = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>AutoGluon training complete, total runtime = 115.13s ... Best model: "WeightedEnsemble_L2"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>TabularPredictor saved. To load, use: predictor = TabularPredictor.load("AutogluonModels\ag-20240804_045523\")</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing feature importance via permutation shuffling for 1605 features using 385 rows with 5 shuffle sets...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    984.4s  = Expected runtime (196.88s per shuffle set)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    325.48s = Actual runtime (Completed 5 of 5 shuffle sets)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                 model  score_test  score_val  pred_time_test  pred_time_val   fit_time  pred_time_test_marginal  pred_time_val_marginal  fit_time_marginal  stack_level  can_infer  fit_order
0             LightGBM    0.633766   0.666667        0.026909       0.014949   9.649190                 0.026909                0.014949           9.649190            1       True          2
1              XGBoost    0.631169   0.677778        0.037873       0.016792  26.016598                 0.037873                0.016792          26.016598            1       True          6
2           ExtraTrees    0.631169   0.616667        0.080732       0.062928   1.251859                 0.080732                0.062928           1.251859            1       True          5
3  WeightedEnsemble_L2    0.631169   0.683333        0.116609       0.080804  30.268527                 0.001993                0.000997           0.413702            2       True          8
4           KNeighbors    0.623377   0.572222        0.052824       0.137540   1.089551                 0.052824                0.137540           1.089551            1       True          1
5         RandomForest    0.610390   0.627778        0.076743       0.063015   3.838227                 0.076743                0.063015           3.838227            1       True          3
6             CatBoost    0.571429   0.666667        0.083691       0.037873  67.730205                 0.083691                0.037873          67.730205            1       True          4
7          LinearModel    0.558442   0.600000        0.078737       0.068869   3.029000                 0.078737                0.068869           3.029000            1       True          7</code></pre>
</div>
</div>
<hr>
</section>
<section id="return" class="level3" data-number="7.4.4">
<h3 data-number="7.4.4" class="anchored" data-anchor-id="return"><span class="header-section-number">7.4.4</span> Return</h3>
<div class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtering the DataFrame to get features where importance &gt; 0 and p_value &lt; 0.05</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>filtered_importance <span class="op">=</span> importance[(importance[<span class="st">'importance'</span>] <span class="op">&gt;</span> <span class="dv">0</span>) <span class="op">&amp;</span> (importance[<span class="st">'p_value'</span>] <span class="op">&lt;</span> <span class="fl">0.05</span>)]</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Counting the number of such features</span></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>number_of_features <span class="op">=</span> filtered_importance.shape[<span class="dv">0</span>]</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Printing the result</span></span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Number of features with importance &gt; 0 and p_value &lt; 0.05: </span><span class="sc">{</span>number_of_features<span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Number of features with importance &gt; 0 and p_value &lt; 0.05: 232</code></pre>
</div>
</div>
<hr>
<div class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the top 20 rows of the Importance DataFrame</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Top 20 rows of Importance DataFrame:"</span>)</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(importance.head(<span class="dv">20</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Top 20 rows of Importance DataFrame:
                importance    stddev   p_value  n  p99_high   p99_low
NTRK1             0.020779  0.012044  0.009090  5  0.045577 -0.004019
RP11-641D5.1      0.018701  0.004996  0.000557  5  0.028989  0.008414
HBA2              0.015065  0.006723  0.003718  5  0.028908  0.001222
STMN2             0.013506  0.005631  0.002917  5  0.025101  0.001912
NAIP              0.012987  0.002597  0.000182  5  0.018335  0.007639
HIST2H2BF         0.012468  0.005631  0.003878  5  0.024062  0.000873
SPP1              0.011948  0.005386  0.003852  5  0.023038  0.000858
AC010524.2        0.011429  0.003939  0.001455  5  0.019539  0.003318
CD24              0.010909  0.002845  0.000508  5  0.016768  0.005051
ADAMDEC1          0.010909  0.006723  0.011097  5  0.024752 -0.002934
XIST              0.010909  0.005631  0.006165  5  0.022503 -0.000685
RP11-1212A22.4    0.010390  0.005808  0.008065  5  0.022348 -0.001569
RPS10-NUDT3       0.009351  0.003939  0.003027  5  0.017461  0.001240
JAKMIP3           0.008831  0.002961  0.001314  5  0.014929  0.002733
PPP1R14C          0.008831  0.002961  0.001314  5  0.014929  0.002733
TBC1D3B           0.008831  0.004718  0.006931  5  0.018546 -0.000884
SPINK5            0.008831  0.001423  0.000078  5  0.011760  0.005902
MMP3              0.008831  0.009293  0.050387  5  0.027965 -0.010303
PPP2R2C           0.008831  0.003939  0.003711  5  0.016942  0.000720
SAA2              0.008312  0.002845  0.001419  5  0.014170  0.002453</code></pre>
</div>
</div>
<hr>
<div class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the top 20 rows of the Leaderboard DataFrame</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">Top 20 rows of Leaderboard DataFrame:"</span>)</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(leaderboard.head(<span class="dv">20</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Top 20 rows of Leaderboard DataFrame:
                 model  score_test  score_val  pred_time_test  pred_time_val  \
0             LightGBM    0.633766   0.666667        0.026909       0.014949   
1              XGBoost    0.631169   0.677778        0.037873       0.016792   
2           ExtraTrees    0.631169   0.616667        0.080732       0.062928   
3  WeightedEnsemble_L2    0.631169   0.683333        0.116609       0.080804   
4           KNeighbors    0.623377   0.572222        0.052824       0.137540   
5         RandomForest    0.610390   0.627778        0.076743       0.063015   
6             CatBoost    0.571429   0.666667        0.083691       0.037873   
7          LinearModel    0.558442   0.600000        0.078737       0.068869   

    fit_time  pred_time_test_marginal  pred_time_val_marginal  \
0   9.649190                 0.026909                0.014949   
1  26.016598                 0.037873                0.016792   
2   1.251859                 0.080732                0.062928   
3  30.268527                 0.001993                0.000997   
4   1.089551                 0.052824                0.137540   
5   3.838227                 0.076743                0.063015   
6  67.730205                 0.083691                0.037873   
7   3.029000                 0.078737                0.068869   

   fit_time_marginal  stack_level  can_infer  fit_order  
0           9.649190            1       True          2  
1          26.016598            1       True          6  
2           1.251859            1       True          5  
3           0.413702            2       True          8  
4           1.089551            1       True          1  
5           3.838227            1       True          3  
6          67.730205            1       True          4  
7           3.029000            1       True          7  </code></pre>
</div>
</div>
<hr>
</section>
<section id="save-data" class="level3" data-number="7.4.5">
<h3 data-number="7.4.5" class="anchored" data-anchor-id="save-data"><span class="header-section-number">7.4.5</span> Save Data</h3>
<div class="cell" data-execution_count="8">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the Importance DataFrame to a CSV file</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>importance.to_csv(<span class="st">'../test_TransProPy/data/Insignificant_correlation_Autogluon_SelectML_importance.csv'</span>, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the Leaderboard DataFrame to a CSV file</span></span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>leaderboard.to_csv(<span class="st">'../test_TransProPy/data/Insignificant_correlation_Autogluon_SelectML_leaderboard.csv'</span>, index<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
</section>
</section>
<section id="significant-correlation-d" class="level2" data-number="7.5">
<h2 data-number="7.5" class="anchored" data-anchor-id="significant-correlation-d"><span class="header-section-number">7.5</span> Significant Correlation</h2>
<blockquote class="blockquote">
<ul>
<li><b style="color: #54467d;">Please note</b>:Data characteristics: Features have strong correlation with the classification.</li>
</ul>
</blockquote>
<section id="import-the-corresponding-module-1" class="level3" data-number="7.5.1">
<h3 data-number="7.5.1" class="anchored" data-anchor-id="import-the-corresponding-module-1"><span class="header-section-number">7.5.1</span> Import the corresponding module</h3>
<div class="cell" data-execution_count="9">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> TransProPy.AutogluonSelectML <span class="im">import</span> AutoGluon_SelectML</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
</section>
<section id="data-1" class="level3" data-number="7.5.2">
<h3 data-number="7.5.2" class="anchored" data-anchor-id="data-1"><span class="header-section-number">7.5.2</span> Data</h3>
<div class="cell" data-execution_count="10">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>data_path <span class="op">=</span> <span class="st">'../test_TransProPy/data/four_methods_degs_intersection.csv'</span>  </span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> pd.read_csv(data_path)</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(data.iloc[:<span class="dv">10</span>, :<span class="dv">10</span>]) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Unnamed: 0  TCGA-D9-A4Z2-01A  TCGA-ER-A2NH-06A  TCGA-BF-A5EO-01A  \
0        A2M         16.808499         16.506184         17.143433   
1      A2ML1          1.584963          9.517669          7.434628   
2      AADAC          4.000000          2.584963          1.584963   
3    AADACL2          1.000000          1.000000          0.000000   
4     ABCA12          4.523562          4.321928          3.906891   
5    ABCA17P          4.584963          5.169925          3.807355   
6      ABCA9          9.753217          6.906891          3.459432   
7      ABCB4          9.177420          6.700440          5.000000   
8      ABCB5         10.134426          4.169925          9.167418   
9     ABCC11         10.092757          6.491853          5.459432   

   TCGA-D9-A6EA-06A  TCGA-D9-A4Z3-01A  TCGA-GN-A26A-06A  TCGA-D3-A3BZ-06A  \
0         17.760739         14.766839         16.263691         16.035207   
1          2.584963          1.584963          2.584963          5.285402   
2          0.000000          0.000000          0.000000          3.321928   
3          0.000000          1.000000          0.000000          0.000000   
4          3.459432          1.584963          3.000000          4.321928   
5          8.366322          7.228819          7.076816          4.584963   
6          2.584963          6.357552          6.475733          7.330917   
7          9.342075         10.392317          7.383704         11.032735   
8          4.906891         11.340963          3.169925         11.161762   
9          6.807355          4.247928          5.459432          5.977280   

   TCGA-D3-A51G-06A  TCGA-EE-A29R-06A  
0         18.355114         16.959379  
1          2.584963          3.584963  
2          1.000000          4.584963  
3          0.000000          1.000000  
4          4.807355          3.700440  
5          6.409391          7.139551  
6          7.954196          9.177420  
7         10.082149         10.088788  
8          4.643856         12.393927  
9          5.614710          8.233620  </code></pre>
</div>
</div>
<hr>
<div class="cell" data-execution_count="11">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>data_path <span class="op">=</span> <span class="st">'../test_TransProPy/data/class.csv'</span>  </span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> pd.read_csv(data_path)</span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(data.iloc[:<span class="dv">10</span>, :<span class="dv">10</span>]) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         Unnamed: 0  class
0  TCGA-D9-A4Z2-01A      2
1  TCGA-ER-A2NH-06A      2
2  TCGA-BF-A5EO-01A      2
3  TCGA-D9-A6EA-06A      2
4  TCGA-D9-A4Z3-01A      2
5  TCGA-GN-A26A-06A      2
6  TCGA-D3-A3BZ-06A      2
7  TCGA-D3-A51G-06A      2
8  TCGA-EE-A29R-06A      2
9  TCGA-D3-A2JE-06A      2</code></pre>
</div>
</div>
<hr>
</section>
<section id="autogluon_selectml-1" class="level3" data-number="7.5.3">
<h3 data-number="7.5.3" class="anchored" data-anchor-id="autogluon_selectml-1"><span class="header-section-number">7.5.3</span> Autogluon_SelectML</h3>
<blockquote class="blockquote">
<ul>
<li>The core purpose of choosing Autogluon_SelectML — to select a larger feature set in AutoGluon that includes both important and secondary features — is reflected in the following custom hyperparameters configuration. This setup is designed to utilize multiple model types so that the models can consider a broader range of features.</li>
<li>This configuration encompasses neural networks (using PyTorch and FastAI), gradient boosting machines (LightGBM, XGBoost, and CatBoost), random forests (RF), extremely randomized trees (XT), K-nearest neighbors (KNN), and linear regression (LR).</li>
</ul>
</blockquote>
<div class="cell" data-execution_count="12">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>importance, leaderboard <span class="op">=</span> AutoGluon_SelectML(</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>    gene_data_path<span class="op">=</span><span class="st">'../test_TransProPy/data/four_methods_degs_intersection.csv'</span>, </span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>    class_data_path<span class="op">=</span><span class="st">'../test_TransProPy/data/class.csv'</span>, </span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>    label_column<span class="op">=</span><span class="st">'class'</span>, </span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>    test_size<span class="op">=</span><span class="fl">0.3</span>, </span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>    threshold<span class="op">=</span><span class="fl">0.9</span>, </span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a>    hyperparameters<span class="op">=</span>{</span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">'GBM'</span>: {}, </span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">'RF'</span>: {},</span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">'CAT'</span>: {}, </span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">'XGB'</span> : {},</span>
<span id="cb104-12"><a href="#cb104-12" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 'NN_TORCH': {}, </span></span>
<span id="cb104-13"><a href="#cb104-13" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 'FASTAI': {},</span></span>
<span id="cb104-14"><a href="#cb104-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">'XT'</span>: {}, </span>
<span id="cb104-15"><a href="#cb104-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">'KNN'</span>: {},</span>
<span id="cb104-16"><a href="#cb104-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">'LR'</span>: {}</span>
<span id="cb104-17"><a href="#cb104-17" aria-hidden="true" tabindex="-1"></a>        },</span>
<span id="cb104-18"><a href="#cb104-18" aria-hidden="true" tabindex="-1"></a>    random_feature<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb104-19"><a href="#cb104-19" aria-hidden="true" tabindex="-1"></a>    num_bag_folds<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb104-20"><a href="#cb104-20" aria-hidden="true" tabindex="-1"></a>    num_stack_levels<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb104-21"><a href="#cb104-21" aria-hidden="true" tabindex="-1"></a>    time_limit<span class="op">=</span><span class="dv">1000</span>, </span>
<span id="cb104-22"><a href="#cb104-22" aria-hidden="true" tabindex="-1"></a>    random_state<span class="op">=</span><span class="dv">42</span></span>
<span id="cb104-23"><a href="#cb104-23" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>No path specified. Models will be saved in: "AutogluonModels\ag-20240804_050246\"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Beginning AutoGluon training ... Time limit = 1000s</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>AutoGluon will save models to "AutogluonModels\ag-20240804_050246\"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>AutoGluon Version:  0.8.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Python Version:     3.10.11</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Operating System:   Windows</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Platform Machine:   AMD64</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Platform Version:   10.0.19044</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Disk Space Avail:   208.48 GB / 925.93 GB (22.5%)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Train Data Rows:    896</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Train Data Columns: 1605</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Label Column: class</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Preprocessing data ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>AutoGluon infers your prediction problem is: 'binary' (because only two unique label-values observed).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    2 unique label values:  [2, 1]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    If 'binary' is not the correct problem_type, please manually specify the problem_type parameter during predictor init (You may specify problem_type as one of: ['binary', 'multiclass', 'regression'])</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Selected class &lt;--&gt; label mapping:  class 1 = 2, class 0 = 1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Note: For your binary classification, AutoGluon arbitrarily selected which label-value represents positive (2) vs negative (1) class.
    To explicitly set the positive_class, either rename classes to 1 and 0, or specify positive_class in Predictor init.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using Feature Generators to preprocess the data ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting AutoMLPipelineFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Available Memory:                    13543.99 MB</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Train Data (Original)  Memory Usage: 11.5 MB (0.1% of available memory)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Inferring data type of each feature based on column values. Set feature_metadata_in to manually specify special dtypes of the features.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Stage 1 Generators:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        Fitting AsTypeFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Stage 2 Generators:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        Fitting FillNaFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Stage 3 Generators:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        Fitting IdentityFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Stage 4 Generators:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        Fitting DropUniqueFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Stage 5 Generators:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        Fitting DropDuplicatesFeatureGenerator...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Types of features in original data (raw dtype, special dtypes):</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        ('float', []) : 1605 | ['A2M', 'A2ML1', 'ABCA12', 'ABCA17P', 'ABCA9', ...]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Types of features in processed data (raw dtype, special dtypes):</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>        ('float', []) : 1605 | ['A2M', 'A2ML1', 'ABCA12', 'ABCA17P', 'ABCA9', ...]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    2.3s = Fit runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1605 features in original data used to generate 1605 features in processed data.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    Train Data (Processed) Memory Usage: 11.5 MB (0.1% of available memory)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Data preprocessing and feature engineering runtime = 2.49s ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>AutoGluon will gauge predictive performance using evaluation metric: 'accuracy'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    To change this, specify the eval_metric parameter of Predictor()</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Automatically generating train/validation split with holdout_frac=0.2, Train Rows: 716, Val Rows: 180</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>User-specified model hyperparameters to be fit:
{
    'GBM': {},
    'RF': {},
    'CAT': {},
    'XGB': {},
    'XT': {},
    'KNN': {},
    'LR': {},
}</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting 7 L1 models ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: KNeighbors ... Training model for up to 997.51s of the 997.49s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.0  = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.24s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.03s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: LightGBM ... Training model for up to 997.21s of the 997.19s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.0  = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.73s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.01s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: RandomForest ... Training model for up to 995.44s of the 995.42s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.0  = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.27s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.05s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: CatBoost ... Training model for up to 994.08s of the 994.06s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.0  = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    65.1s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.04s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: ExtraTrees ... Training model for up to 928.91s of the 928.89s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.0  = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.11s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.05s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: XGBoost ... Training model for up to 927.71s of the 927.69s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.0  = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    4.56s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.02s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: LinearModel ... Training model for up to 923.11s of the 923.09s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>E:\Anaconda\Anaconda\envs\TransPro\lib\site-packages\sklearn\preprocessing\_data.py:2663: UserWarning:

n_quantiles (1000) is greater than the total number of samples (716). n_quantiles is set to n_samples.
</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.0  = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    2.78s    = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.06s    = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Fitting model: WeightedEnsemble_L2 ... Training model for up to 360.0s of the 920.19s of remaining time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    1.0  = Validation score   (accuracy)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.4s     = Training   runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    0.0s     = Validation runtime</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>AutoGluon training complete, total runtime = 80.26s ... Best model: "WeightedEnsemble_L2"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>TabularPredictor saved. To load, use: predictor = TabularPredictor.load("AutogluonModels\ag-20240804_050246\")</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing feature importance via permutation shuffling for 1605 features using 385 rows with 5 shuffle sets...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    640.26s = Expected runtime (128.05s per shuffle set)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    65.47s  = Actual runtime (Completed 5 of 5 shuffle sets)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                 model  score_test  score_val  pred_time_test  pred_time_val   fit_time  pred_time_test_marginal  pred_time_val_marginal  fit_time_marginal  stack_level  can_infer  fit_order
0             LightGBM    1.000000        1.0        0.016944       0.011959   1.733270                 0.016944                0.011959           1.733270            1       True          2
1           KNeighbors    1.000000        1.0        0.050829       0.033886   0.238504                 0.050829                0.033886           0.238504            1       True          1
2         RandomForest    1.000000        1.0        0.070763       0.049834   1.269384                 0.070763                0.049834           1.269384            1       True          3
3           ExtraTrees    1.000000        1.0        0.070764       0.052823   1.109194                 0.070764                0.052823           1.109194            1       True          5
4          LinearModel    1.000000        1.0        0.071759       0.063786   2.781226                 0.071759                0.063786           2.781226            1       True          7
5  WeightedEnsemble_L2    1.000000        1.0        0.072757       0.053820   1.508564                 0.001993                0.000997           0.399371            2       True          8
6             CatBoost    0.997403        1.0        0.084717       0.036279  65.102499                 0.084717                0.036279          65.102499            1       True          4
7              XGBoost    0.994805        1.0        0.031897       0.015007   4.556480                 0.031897                0.015007           4.556480            1       True          6</code></pre>
</div>
</div>
<hr>
</section>
<section id="return-1" class="level3" data-number="7.5.4">
<h3 data-number="7.5.4" class="anchored" data-anchor-id="return-1"><span class="header-section-number">7.5.4</span> Return</h3>
<div class="cell" data-execution_count="13">
<div class="sourceCode cell-code" id="cb190"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb190-1"><a href="#cb190-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtering the DataFrame to get features where importance &gt; 0 and p_value &lt; 0.05</span></span>
<span id="cb190-2"><a href="#cb190-2" aria-hidden="true" tabindex="-1"></a>filtered_importance <span class="op">=</span> importance[(importance[<span class="st">'importance'</span>] <span class="op">&gt;</span> <span class="dv">0</span>) <span class="op">&amp;</span> (importance[<span class="st">'p_value'</span>] <span class="op">&lt;</span> <span class="fl">0.05</span>)]</span>
<span id="cb190-3"><a href="#cb190-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Counting the number of such features</span></span>
<span id="cb190-4"><a href="#cb190-4" aria-hidden="true" tabindex="-1"></a>number_of_features <span class="op">=</span> filtered_importance.shape[<span class="dv">0</span>]</span>
<span id="cb190-5"><a href="#cb190-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Printing the result</span></span>
<span id="cb190-6"><a href="#cb190-6" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="ss">f"Number of features with importance &gt; 0 and p_value &lt; 0.05: </span><span class="sc">{</span>number_of_features<span class="sc">}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Number of features with importance &gt; 0 and p_value &lt; 0.05: 0</code></pre>
</div>
</div>
<hr>
<div class="cell" data-execution_count="14">
<div class="sourceCode cell-code" id="cb192"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb192-1"><a href="#cb192-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the top 20 rows of the Importance DataFrame</span></span>
<span id="cb192-2"><a href="#cb192-2" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Top 20 rows of Importance DataFrame:"</span>)</span>
<span id="cb192-3"><a href="#cb192-3" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(importance.head(<span class="dv">20</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Top 20 rows of Importance DataFrame:
          importance  stddev  p_value  n  p99_high  p99_low
A2M              0.0     0.0      0.5  5       0.0      0.0
PROM2            0.0     0.0      0.5  5       0.0      0.0
PSORS1C2         0.0     0.0      0.5  5       0.0      0.0
PSORS1C1         0.0     0.0      0.5  5       0.0      0.0
PSMC1P1          0.0     0.0      0.5  5       0.0      0.0
PRTG             0.0     0.0      0.5  5       0.0      0.0
PRSS8            0.0     0.0      0.5  5       0.0      0.0
PRSS53           0.0     0.0      0.5  5       0.0      0.0
PRSS3            0.0     0.0      0.5  5       0.0      0.0
PRSS22           0.0     0.0      0.5  5       0.0      0.0
PRR19            0.0     0.0      0.5  5       0.0      0.0
PRR15L           0.0     0.0      0.5  5       0.0      0.0
PRODH            0.0     0.0      0.5  5       0.0      0.0
PI16             0.0     0.0      0.5  5       0.0      0.0
PRKCQ            0.0     0.0      0.5  5       0.0      0.0
PRF1             0.0     0.0      0.5  5       0.0      0.0
PRAME            0.0     0.0      0.5  5       0.0      0.0
PPP2R2C          0.0     0.0      0.5  5       0.0      0.0
PPP1R1B          0.0     0.0      0.5  5       0.0      0.0
PPP1R14C         0.0     0.0      0.5  5       0.0      0.0</code></pre>
</div>
</div>
<hr>
<div class="cell" data-execution_count="15">
<div class="sourceCode cell-code" id="cb194"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb194-1"><a href="#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the top 20 rows of the Leaderboard DataFrame</span></span>
<span id="cb194-2"><a href="#cb194-2" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">Top 20 rows of Leaderboard DataFrame:"</span>)</span>
<span id="cb194-3"><a href="#cb194-3" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(leaderboard.head(<span class="dv">20</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Top 20 rows of Leaderboard DataFrame:
                 model  score_test  score_val  pred_time_test  pred_time_val  \
0             LightGBM    1.000000        1.0        0.016944       0.011959   
1           KNeighbors    1.000000        1.0        0.050829       0.033886   
2         RandomForest    1.000000        1.0        0.070763       0.049834   
3           ExtraTrees    1.000000        1.0        0.070764       0.052823   
4          LinearModel    1.000000        1.0        0.071759       0.063786   
5  WeightedEnsemble_L2    1.000000        1.0        0.072757       0.053820   
6             CatBoost    0.997403        1.0        0.084717       0.036279   
7              XGBoost    0.994805        1.0        0.031897       0.015007   

    fit_time  pred_time_test_marginal  pred_time_val_marginal  \
0   1.733270                 0.016944                0.011959   
1   0.238504                 0.050829                0.033886   
2   1.269384                 0.070763                0.049834   
3   1.109194                 0.070764                0.052823   
4   2.781226                 0.071759                0.063786   
5   1.508564                 0.001993                0.000997   
6  65.102499                 0.084717                0.036279   
7   4.556480                 0.031897                0.015007   

   fit_time_marginal  stack_level  can_infer  fit_order  
0           1.733270            1       True          2  
1           0.238504            1       True          1  
2           1.269384            1       True          3  
3           1.109194            1       True          5  
4           2.781226            1       True          7  
5           0.399371            2       True          8  
6          65.102499            1       True          4  
7           4.556480            1       True          6  </code></pre>
</div>
</div>
<hr>
</section>
<section id="save-data-1" class="level3" data-number="7.5.5">
<h3 data-number="7.5.5" class="anchored" data-anchor-id="save-data-1"><span class="header-section-number">7.5.5</span> Save Data</h3>
<div class="cell" data-execution_count="16">
<div class="sourceCode cell-code" id="cb196"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb196-1"><a href="#cb196-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the Importance DataFrame to a CSV file</span></span>
<span id="cb196-2"><a href="#cb196-2" aria-hidden="true" tabindex="-1"></a>importance.to_csv(<span class="st">'../test_TransProPy/data/significant_correlation_Autogluon_SelectML_importance.csv'</span>, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb196-3"><a href="#cb196-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb196-4"><a href="#cb196-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the Leaderboard DataFrame to a CSV file</span></span>
<span id="cb196-5"><a href="#cb196-5" aria-hidden="true" tabindex="-1"></a>leaderboard.to_csv(<span class="st">'../test_TransProPy/data/significant_correlation_Autogluon_SelectML_leaderboard.csv'</span>, index<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
</section>
</section>
<section id="references-d" class="level2" data-number="7.6">
<h2 data-number="7.6" class="anchored" data-anchor-id="references-d"><span class="header-section-number">7.6</span> References</h2>
<section id="scientific-publications" class="level3" data-number="7.6.1">
<h3 data-number="7.6.1" class="anchored" data-anchor-id="scientific-publications"><span class="header-section-number">7.6.1</span> Scientific Publications</h3>
<blockquote class="blockquote">
<ul>
<li><p><a href="https://arxiv.org/abs/2003.06505" style="color: #43799a; font-weight: bold;">Erickson, N., Mueller, J., Shirkov, A., Zhang, H., Larroy, P., Li, M., &amp; Smola, A. (2020). AutoGluon-Tabular: Robust and Accurate AutoML for Structured Data. arXiv preprint arXiv:2003.06505.</a></p></li>
<li><p><a href="https://arxiv.org/abs/2006.14284" style="color: #43799a; font-weight: bold;">Fakoor, R., Mueller, J., Erickson, N., Chaudhari, P., &amp; Smola, A. J. (2020). Fast, Accurate, and Simple Models for Tabular Data via Augmented Distillation. arXiv preprint arXiv:2006.14284. </a></p></li>
<li><p><a href="https://openreview.net/forum?id=OHAIVOOl7Vl" style="color: #43799a; font-weight: bold;">Shi, X., Mueller, J., Erickson, N., Li, M., &amp; Smola, A. (2021). Multimodal AutoML on Structured Tables with Text Fields. In AutoML@ICML 2021. </a></p></li>
</ul>
</blockquote>
</section>
<section id="articles" class="level3" data-number="7.6.2">
<h3 data-number="7.6.2" class="anchored" data-anchor-id="articles"><span class="header-section-number">7.6.2</span> Articles</h3>
<blockquote class="blockquote">
<ul>
<li><p><a href="https://aws.amazon.com/blogs/opensource/machine-learning-with-autogluon-an-open-source-automl-library/" style="color: #43799a; font-weight: bold;">Prasanna, S. (2020, March 31). Machine learning with AutoGluon, an open source AutoML library. AWS Open Source Blog. </a></p></li>
<li><p><a href="https://medium.com/@zhanghang0704/image-classification-on-kaggle-using-autogluon-fc896e74d7e8" style="color: #43799a; font-weight: bold;">Sun, Y., Wu, C., Zhang, Z., He, T., Mueller, J., &amp; Zhang, H. (n.d.). (2020). Image classification on Kaggle using AutoGluon. Medium.</a></p></li>
<li><p><a href="https://towardsdatascience.com/autogluon-deep-learning-automl-5cdb4e2388ec" style="color: #43799a; font-weight: bold;">Erickson, N., Mueller, J., Zhang, H., &amp; Kamakoti, B. (2019). AutoGluon: Deep Learning AutoML. Medium.</a></p></li>
</ul>
</blockquote>
</section>
<section id="documentation" class="level3" data-number="7.6.3">
<h3 data-number="7.6.3" class="anchored" data-anchor-id="documentation"><span class="header-section-number">7.6.3</span> Documentation</h3>
<blockquote class="blockquote">
<ul>
<li><a href="https://auto.gluon.ai/0.1.0/api/autogluon.predictor.html?highlight=num_bag_folds" style="color: #43799a; font-weight: bold;">AutoGluon Predictors –AutoGluon Documentation 0.1.0 documentation</a></li>
</ul>
</blockquote>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://giscus.app/client.js" data-repo="SSSYDYSSS/TransProPyBook" data-repo-id="R_kgDOKxwgVA" data-category="General" data-category-id="DIC_kwDOKxwgVM4CbO4f" data-mapping="title" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="light" data-lang="en" crossorigin="anonymous" async="">
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./AutogluonTimeLimit.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">AutogluonTimeLimit.py</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./AutoFeatureSelection.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">AutoFeatureSelection.py</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>